src/WebJobs.Script/Workers/Rpc/RpcWorkerConstants.cs

```diff
@@ -55,6 +55,11 @@ public static class RpcWorkerConstants
         public const string HandlesWorkerTerminateMessage = "HandlesWorkerTerminateMessage";
         public const string HandlesInvocationCancelMessage = "HandlesInvocationCancelMessage";
 
+        /// <summary>
+        /// Indicates whether empty entries in the trigger message should be included when sending RpcInvocation data to OOP workers.
+        /// </summary>
+        public const string IncludeEmptyEntriesInMessagePayload = "IncludeEmptyEntriesInMessagePayload";
```

Comment:
This is a new capability and none of the worker are sending it now. We will start with dotnet worker and then share the info with other language workers so that they can also opt in for this behavior.

Reply:
Are you saying that if we simply updated the ToRpcStringArray method to correctly add nulls with full fidelity, existing workers won't be able to handle that? Can you link to the corresponding worker code showing where the null-ref occurs?

Reply:
> Are you saying that if we simply updated the ToRpcStringArray method to correctly add nulls with full fidelity, existing workers won't be able to handle that? Can you link to the corresponding worker code showing where the null-ref occurs?

No. I did not mean that.  The fix is correcting the current incorrect behavior/bug. So probably do not need to be considered as a breaking change or so. I will remove the capability check.

Reply:
Latest iteration has this capability check removed.

Reply:
So we'll now send along empty entries, but still skip nulls. For empty entries, are we sure workers handle this w/o breaking changes? Probably need to verify with some testing for the triggers supporting cardinality many (EventHubs, ServiceBus Batches).

This will address the linked issue (empty strings) but what about nulls? Is it possible for the triggers to send nulls here? If so, in those cases, we'll have the same issue.

Reply:
null cannot be added as `RepeatedField<string>.Add` will throw when adding a null entry.

The original idea of having the capability check was to prevent this being a breaking change for workers when host version update (Even though the change is to fix a bug/incorrect behavior). Do you think we should go back to that route to be safe/reduce potential impact? I am assuming none of the worker implementations having specific code to handle the current incorrect data (without empty entries) coming in. If they do, we bite the bullet with this change(host version change) and fix it now? 

I tested with EventHubs batching use case in dotnet isolated worker and it is behaving correctly.(the original incorrect behavior issue was reported in dotnet worker as mentioned

Reply:
If we can verify that for all of the language workers as they exist now they correctly handle the new empty array elements we may send them without blowing up, then we don't need a capability flag I think.

Reply:
Is it really that this cannot be null? We are using `Microsoft.Azure.Devices.Client.Message` without a body and only with properties. That IoTHub message is then routed to this eventhub. Is the default body of an IoT Hub message `string.Empty`?

Reply:
Language worker testing is currently blocked due to [this issue](https://github.com/Azure/azure-functions-python-worker/issues/1089) on python.  Tested JS, TS and Dotnetworker and they look good.

Reply:
Historically, any behavior change like this is based on capability advertisement. That’s the safest way if you want to expedite the fix. If you don’t want to introduce a capability, thorough testing of all language workers will be required, and that wouldn’t cover external ones other teams may be maintaining.

is there a concern about the new capability?

Reply:
Brought back the capability check based on offline sync. Thank you!

Reply:
We are now testing this bugfix in our azure function but it looks like we are still hitting this issue. Nobody also replied to my comment above...

> Is it really that this cannot be null? We are using `Microsoft.Azure.Devices.Client.Message` without a body (there is only constructor where you don't need to specify a body). We only set properties and a messageId on this Message. That IoTHub message is then routed to this eventhub. Is the default body of an IoT Hub message `string.Empty`?

It looks like our payload was just null and this bugfix is only helping when you are sending string.Empty. Which `Microsoft.Azure.Devices.Client.Message` doesn't do by default.

CC @kshyju, @fabiocav